packages/go_router/lib/src/parser.dart

```diff
@@ -86,7 +86,8 @@ class GoRouteInformationParser extends RouteInformationParser<RouteMatchList> {
         // to uri, empty path check might be required here; see
         // https://github.com/flutter/packages/pull/5113#discussion_r1374861070
```

Comment:
The entire todo and ignores above can be removed, and we need to do empty path check here as well per comment above

Reply:
> The entire todo and ignores above can be removed, and we need to do empty path check here as well per comment above

I am not sure that we necessarily need to do the empty path check. It seems that you feel we need to do this check so that we do not re-introduce [this issue](https://github.com/flutter/flutter/issues/133928). But when launching our company's app through a deep link with no path (https://www.wwt.com) from a "dead" state, we are not seeing this `GoException: no routes for location` error message and the app launches as expected.

Maybe you can check to see if this issue has been re-introduced on your side?

Reply:
Let me check and I will get back to you

Reply:
I am testing in android, it looks like when cold start without the path the uri doesn't contain the host name and scheme. I am not sure what happen here.

If I warm start without the path, it would show no route for location. It seems to me we need to handle empty path here.

Reply:
Yeah, I am actually seeing that we get the no route for location error when launching `www.domain.com` while the app is already running in the background.

It's almost as if we need to treat all `www.domain.com` URIs as `www.domain.com/` so that the location defaults to `/`. It makes sense that having a location of an empty string wouldn't match anything.

Reply:
@chunhtai Does this sound like the right approach to you?

Reply:
@chunhtai we went ahead and added the empty path check to ensure that `routeInformation` with an empty `.uri.path` has a `/` appended to the end of it. This behavior matches [what we had previously been doing with `routeInformation.location`](https://github.com/flutter/flutter/blob/b62bb518f4ce2a2df6ca7abec74e73629b61aab5/packages/flutter/lib/src/widgets/router.dart#L66-L77).